### PR TITLE
Android basedir

### LINKED ISSUE #11633: Android runtime error
The [current](https://github.com/servo/servo/commit/1bc94c132e6820293c2178bf95991518b1e8a677) android release is failing with the following error:
`D/RustAndroidGlueStdouterr(18547): thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: MissingHomeVariable', ../src/libcore/result.rs:785`

### PR DESCRIPTION
**Pull request by larsbergstrom**
This fixes #11633, as when the new basedir stuff that uses xdg was added, it did not preserve the previous Android behavior, which used `/sdcard/servo`. Obviously, long-term we should be calling into the Android runtime to determine the per-run installation path and be both unpacking and loading resources from there, but this restores the previous behavior.

r? @mbrubeck 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #11633 (github issue number if applicable).

- [x] There are tests for these changes OR
- [x] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11687)

### CODE REVIEW

**Comment by larsbergstrom** [comment]
The added libz update fixes #11627

**Comment by mbrubeck** [comment]
@bors-servo retry

* #11693

File: support/android/apk/build.xml [hunk 1]
```diff
@@ -36,9 +36,17 @@
...
+    <!-- The project.properties file is created and updated by the 'android'
+         tool, as well as ADT.
+
+         This contains project specific properties such as project target, and library
+         dependencies. Lower level build properties are stored in ant.properties
+         (or in .classpath for Eclipse projects).
+
+         This file is an integral part of the build system for your
+         application and should be checked into Version Control Systems. -->
+    <loadproperties srcFile="project.properties" />
```
Comment (aneeshusa) [inline support/android/apk/build.xml]: I believe @mitchhentges removed this file in #11278. If the Android automated tooling rewrote the build.xml file to link to it, then we should re-add the project.properties file. If you changed build.xml by hand, I think we can revert that change.

Comment (larsbergstrom) [inline support/android/apk/build.xml]: Yeah, the automated tooling will always put that file back in there. I *thought* I had argued against that change, but clearly I didn't. Oops!

**Pull request merged**